OneCommunityGlobal/HighestGoodNetworkApp
# Obeda fix another person view dashboard view header

# Description
![task](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/61763829/fab3e96c-f6fa-4a59-ac53-c9fab35a1aac)


## Related PRS (if any):
Reverted [#991](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/991)
…

## Main changes explained:
- The Header.jsx always displayed the header of the logged in account even when viewing another account because it used data from props.auth, which contains data of the logged in user
- In Header.jsx, replaced props.auth to props.userProfile and made minor adjustments according to it so that the header of the viewed account is displayed
- Likewise, in the Permission file, used state.auth instead of state.userProfile

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log in with different user roles (read note below)
4. Leaderboard -> Click dot by person's name -> Click OK
5. Check the header section to ensure that: 
	a. the correct name of the viewed account is displayed
	b. the correct menu options are displayed according to their account type 
	c. the menu options work properly

## Screenshots or videos of changes:
Before:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/61763829/03ea6aed-b6b9-4300-b58e-60f8a456965e

After:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/61763829/cd52ae9b-034a-4ab7-9d93-2ca75a0f6b43



## Note:
 1. Admin/Owner account should be able to interact with other accounts
 2. Non admin/owner account should not be able to interact with other person's header options
